### PR TITLE
Use NAD parameters to configure the Octavia resources

### PR DESCRIPTION
**Pull request by gthiemonge**
JIRA: [OSPRH-7553](https://issues.redhat.com//browse/OSPRH-7553)

### CODE REVIEW

File: pkg/octavia/lb_mgmt_network.go [hunk 1]
```diff
@@ -63,43 +64,31 @@ func findPort(client *gophercloud.ServiceClient, networkID string, ipAddress str
...
-	p, err := findPort(client, tenantNetwork.ID, ipAddress, log)
+func ensurePort(client *gophercloud.ServiceClient, tenantNetwork *networks.Network, securityGroups *[]string, log *logr.Logger) (*ports.Port, error) {
+	p, err := findPort(client, tenantNetwork.ID, LbMgmtRouterPortName, log)
 	if err != nil {
 		return nil, err
 	}
 	if p != nil {
 		//
-		// TODO(beagles): reconcile port properties? Is there anything to do? Security groups possibly.
+		// TODO(beagles): reconcile port properties? Is there anything to do? Secur/Hostity groups possibly.
```
Comment (weinimo) [inline pkg/octavia/lb_mgmt_network.go:82]: Was that an accidental change?

File: pkg/octavia/network_parameters.go [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+
+	nadConfig, err := getConfigFromNAD(nad)
+	if err != nil {
+		return nil, fmt.Errorf("cannot read network parameters: %w", err)
+	}
+
+	// Provider subnet parameters
+	networkParameters.ProviderCIDR = nadConfig.IPAM.CIDR
+
+	networkParameters.ProviderAllocationStart = nadConfig.IPAM.RangeEnd.Next()
```
Comment (weinimo) [inline pkg/octavia/network_parameters.go:94]: Can you describe what this line, and maybe the once till line 102 is doing? It seems it checks if the IP ranges in given `networkParameters` are within the lb provider subnet, but the algorithm isn't quite clear to me.

**Review by weinimo — CHANGES_REQUESTED** [review]
found only minor issues. A few more comments wouldn't hurt.

File: pkg/octavia/lb_mgmt_network.go (hunk 1 above)
Comment (gthiemonge) [reply pkg/octavia/lb_mgmt_network.go:82]: It was. Fixed

**Review by weinimo — APPROVED** [review]

**Pull request merged**